ddl/db_change_test.go

```diff
@@ -1826,25 +1830,88 @@ func (s *serialTestStateChangeSuite) TestCreateExpressionIndex(c *C) {
 		c.Assert(err, IsNil)
 		switch job.SchemaState {
 		case model.StateDeleteOnly:
-			_, checkErr = tk1.Exec("insert into t values (5, 5)")
-			if checkErr != nil {
-				return
+			for _, sql := range stateDeleteOnlySQLs {
+				_, checkErr = tk1.Exec(sql)
+				if checkErr != nil {
+					return
+				}
+			}
+			// (1, 7), (2, 2), (3, 3), (5, 5), (0, 6)
+		case model.StateWriteOnly:
+			for _, sql := range stateWriteOnlySQLs {
+				_, checkErr = tk1.Exec(sql)
+				if checkErr != nil {
+					return
+				}
 			}
-			_, checkErr = tk1.Exec("insert into t set b = 6")
+			// (1, 7), (2, 7), (5, 5), (0, 6), (8, 8), (0, 9)
+		case model.StateWriteReorganization:
+			for _, sql := range stateWriteReorganizationSQLs {
+				_, checkErr = tk1.Exec(sql)
+				if checkErr != nil {
+					return
+				}
+			}
+			// (1, 7), (2, 7), (5, 7), (8, 8), (0, 9), (10, 10), (10, 10), (0, 11), (0, 11)
+		}
+	}
+
+	d.(ddl.DDLForTest).SetHook(callback)
+	tk.MustExec("alter table t add index idx((b+1))")
+	c.Assert(checkErr, IsNil)
+	tk.MustExec("admin check table t")
```

Comment:
how about adding `admin check table t` at the end of each state.

Reply:
It's useless because the index is not public yet.